src/esssans/normalization.py

```diff
@@ -64,39 +64,63 @@ def solid_angle_rectangular_approximation(
     )
 
 
-def solid_angle_approximation_for_cylinders(cylinders: sc.DataSet, origin: sc.Variable):
-    """Approximation of the solid angle of a cylinder as seen from `origin`.
-    The approximation does not take edge caps into account.
-
-    Cylinder is approximated by a rectangle cross section parallel
-    to the axis of the cylinder.
-    The rectangle cross section is selected to maximize the solid angle.
-
-    The solid angle of the rectangular cross section is computed by
-    splitting it into two triangles and applying the
-    solid angle equation for triangles in 3D,
-    source: doi.org/10.1080/0025570X.1990.11977515.
+def solid_angle_rectangular_approximation_cylinder(
+    detector: DetectorGeometry[RunType],
+) -> SolidAngle[RunType]:
+    """
+    Solid angle computed from cylindrical pixels.
+
+    Note that this approximation is only valid when the distance between sample and pixel is much larger than the length or the radius of the pixel.
+
+    Parameters
+    ----------
+    detector:
+        The DataGroup that contains the geometry of the detector. Must contain pixel_shape, the transformation to the sample reference frame and the position (center) of each pixel.
+
+    Returns
+    -------
+    :
+        The solid angle of the detector pixels, as viewed from the sample position.
+    """
+
+
+    face_1_center, face_1_edge, face_2_center = (
+        detector['pixel_shape']['vertices']['vertex', i]
+        for i in range(3)
+    )
+    cylinder_axis = (
+        detector['transform'] * face_2_center -
+        detector['transform'] * face_1_center
+    )
+    radius = sc.norm(face_1_center - face_1_edge)
+    length = sc.norm(cylinder_axis)
+
+    return SolidAngle[RunType](
+        approximate_solid_angle_for_cylinder_shaped_pixel_of_detector(
+            # Is there a way to get the positions of the pixels without going through the events?
```

Comment:
You should assume that the input is a `DataArray` with a `position` coord, not the parent data group. This may be containing events or not. For the lowest-level function it may make sense to just take the position.

Reply:
Should I assume the data array also has the `pixel_shape` data or should that enter from a separate provider?

Reply:
The array does not have pixel shape, since it is stored in a `DataGroup`, on a higher level, i.e., use a separate provider.